pkg/tagset/null_factory_test.go

```diff
@@ -0,0 +1,121 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.Datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// NOTE: this file is named *_test.go because it is only intended for use in
+// tests within this package.
+
+package tagset
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/twmb/murmur3"
+)
+
+func TestNullFactory(t *testing.T) {
+	testFactory(t, func() Factory { return newNullFactory() })
+}
+
+// A nullFactory caches nothing.
+//
+// This type implements Factory.
```

Comment:
Maybe better to have this instead of a comment
```go
var _ Factory = (*nullFactory)(nil)
```

I think it leads to cleaner errors and it also serves as documentation. (Maybe even a good idea to have this on every `Factory` implementation actually)

Reply:
Oh, good point!